### DISCUSSION TITLE
Recently sold tesla, all data from teslamate is gone

### DISCUSSION THREAD

**Discussion by markup26** [teslamate-org/teslamate#3938] [Q&A] [answered]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

today I sold my MY, as soon as I removed the car from the Tesla app, my car disappeared from teslamate and grafana. I see my car is still on the database under the cars table. Is there a way of getting all the data to show up on Grafana for historical reference?

### Relevant log output

```shell
N/A
```

### Type of installation

Docker

### Version

1.29.1

**Comment by cwanja** [comment]
Moving to discussion as this is not a bug / issue with TeslaMate.

**Comment by cwanja** [comment]
What is the error in Grafana when you view a dashboard? With the data still in the database, your dashboards should continue to work.

**Reply by markup26** [reply]
On grafana, the drop down option to select which car to see, only shows the two other cars in my account, but not my old car.

**Reply by markup26** [reply, marked as answer]
I was able to fix the issue, it seems that I had two cars on the database with no name, so Grafana was not picking up the 2nd car with no name. I added the name to the database entry for my old car and everything is now back in Grafana.

**Reply by cwanja** [reply]
Glad to hear it @markup26!

**Reply by JakobLichterfeld** [reply]
See docs as well: https://docs.teslamate.org/docs/faq#why-null-is-displayed-above-the-panels-in-grafana
